### PR TITLE
update of the script for release validation

### PR DESCRIPTION
**Pull request by emeninno**
Dear all,

I hope I implemented all the comments from Chiara and Benedikt from the previous PR. This is the updated version of the macro for release validation.
If I forgot something just tell me.
Cheers,
Elisa

### CODE REVIEW

File: RelVal/ReleaseValidation.C [hunk 1]
```diff
@@ -20,16 +20,16 @@ struct results
...
 };
 
-// define "a monte" the possible available tests
+// define the possible available tests
 enum options {
   Chi2    = 0x01,
-  BinCont     = 0x02,
-  BinContInt   = 0x04, //BinContInt is similar to BinCont, but the bin content is normalized by the integral 
+  BinContNorm     = 0x02,
+  Nentries = 0x04,
```
Comment (chiarazampolli) [inline RelVal/ReleaseValidation.C:27]: Shouldn't enums be fully capital?

File: RelVal/ReleaseValidation.C [hunk 2]
```diff
@@ -345,45 +346,66 @@ void WriteHisto2D(TObject *obj) {
...
 void WriteTEfficiency(TObject *obj) { // should I further develop that?
+
   TEfficiency *hEff = static_cast<TEfficiency *>(obj);
+
+  // separate numerator and denominator of the efficiency
   TH1 *hEffNomin  = (TH1*)hEff->GetPassedHistogram(); // eff nominator
   TH1 *hEffDenom = (TH1*)hEff->GetTotalHistogram();   // eff denominator
+
+  // recreate the efficiency dividing numerator for denominator:
+  TH1* heff = dynamic_cast<TH1*>(hEffNomin->Clone("heff"));
```
Comment (chiarazampolli) [inline RelVal/ReleaseValidation.C:357]: I would not use a dynamic_cast, just `TH1* heff = (TH1*)(hEffNomin->Clone("heff"));`

File: RelVal/ReleaseValidation.C [hunk 3]
```diff
@@ -345,45 +346,66 @@ void WriteHisto2D(TObject *obj) {
...
+  heff->Divide(hEffNomin, hEffDenom, 1.0, 1.0, "B");
   TDirectory *current = gDirectory;
 
   // save nominator and denominator of the efficiency, to compare these plots from the two input files
-  TCanvas *cc = new TCanvas(Form("%s_%s", fileOut->GetName(),hEff->GetName()),Form("%s_%s", fileOut->GetName(),hEff->GetName()));
+
+  TCanvas *cc = new TCanvas("Efficiency",Form("%s_%s", fileOut->GetName(),hEff->GetName()));
```
Comment (chiarazampolli) [inline RelVal/ReleaseValidation.C:363]: Add spaces after commas (everywhere in the PR where it is not done):
```
TCanvas *cc = new TCanvas("Efficiency", Form("%s_%s", fileOut->GetName(), hEff->GetName()));
```

File: RelVal/ReleaseValidation.C [hunk 4]
```diff
@@ -345,45 +346,66 @@ void WriteHisto2D(TObject *obj) {
...
 
   // save nominator and denominator of the efficiency, to compare these plots from the two input files
-  TCanvas *cc = new TCanvas(Form("%s_%s", fileOut->GetName(),hEff->GetName()),Form("%s_%s", fileOut->GetName(),hEff->GetName()));
+
+  TCanvas *cc = new TCanvas("Efficiency",Form("%s_%s", fileOut->GetName(),hEff->GetName()));	       
   hEff->Draw("AP");
-  TCanvas *cnom = new TCanvas(Form("%s_%s_nomin", fileOut->GetName(),hEffNomin->GetName()),Form("%s_%s_nomin", fileOut->GetName(),hEffNomin->GetName()));
+  cc->SaveAs(Form("%s_%s.png", fileOut->GetName(),hEff->GetName()));
+  
+  TCanvas *cnom = new TCanvas("eff numerator",Form("%s_%s_effnominator", fileOut->GetName(),hEffNomin->GetName()));
```
Comment (chiarazampolli) [inline RelVal/ReleaseValidation.C:367]: space after comma

File: RelVal/ReleaseValidation.C [hunk 5]
```diff
@@ -345,45 +346,66 @@ void WriteHisto2D(TObject *obj) {
...
   hEffNomin->Draw();
-  cnom->SaveAs(Form("%s_%s_nomin.png", fileOut->GetName(),hEffNomin->GetName()));
-  TCanvas *cden = new TCanvas(Form("%s_%s_nomin", fileOut->GetName(),hEffDenom->GetName()),Form("%s_%s_denom", fileOut->GetName(),hEffDenom->GetName()));
+  cnom->SaveAs(Form("%s_%s_effnominator.png", fileOut->GetName(),hEffNomin->GetName()));
```
Comment (chiarazampolli) [inline RelVal/ReleaseValidation.C:369]: space after comma

File: RelVal/ReleaseValidation.C [hunk 6]
```diff
@@ -459,34 +481,34 @@ void CompareHistos(TH1 *hA, TH1 *hB, TString monobj, int whichTest, double valCh
       hTests->Fill("Chi2 test", Form("%s",hA->GetName()),1); // GOOD--> histo bin cont = 0 
     }      
   }
-  if ( (whichTest & BinCont) == BinCont) {
-    testResult = CompareChiSquareBinContentNentr(hA, hB, BinCont , valChi2, valMeanDiff);
+  if ( (whichTest & BinContNorm) == BinContNorm) {
```
Comment (chiarazampolli) [inline RelVal/ReleaseValidation.C:484]: `( (` should be `((`

**Comment by emeninno** [comment]
> @emeninno Have you applied clang-format? I feel like many comments by @chiarazampolli should be fixed by that. As far as I know I sent you the instructions via mail. Let me know if you need more info here. In case you have difficulties, I could make a PR to your branch with clang-format applied and after that you can address the other comments here.

Hi Benedikt, yes I followed your instrusctions for the clang-format (thanks a lot for the email with instructions). 
I now implemented the comments by Chiara, let me try again to apply the clang-format and then I can push again

**Comment by benedikt-voelkel** [comment]
Failing CI unrelated. @chiarazampolli, I think if there are no further comments, this PR should be good to be merged
